src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<? extends Machine> bulldozers = new BulldozerProducer().get();
+            return new ArrayList<>(bulldozers);
+        }
+        if (type == Excavator.class) {
+            List<? extends Machine> excavators = new ExcavatorProducer().get();
+            return new ArrayList<>(excavators);
+        }
+        if (type == Truck.class) {
+            List<? extends Machine> trucks = new TruckProducer().get();
+            return new ArrayList<>(trucks);
+        }
+        return new ArrayList<>();
```

Comment:
This code must be a good example of polymorphism.
Let's create an abstract variable of a type MachineProducer and initialize it with an applicable producer implementation. Then just call the method get one time.

------
(optional) Also, you can try to use a hashMap instead of if-blocks. It is a really good practice.